src/util/WindowsHelper.cpp

```diff
@@ -83,6 +88,54 @@ void setRegisteredForStartup(bool isRegistered)
     }
 }
 
+QString getAssociatedCommand(ASSOCIATION_QUERY_TYPE queryType, LPCWSTR query)
+{
+    static HINSTANCE shlwapi = LoadLibrary(L"shlwapi");
+    static auto assocQueryString = AssocQueryString_(
+        (shlwapi == NULL) ? NULL
+                          : GetProcAddress(shlwapi, "AssocQueryStringW"));
+
+    if (assocQueryString == NULL)
+    {
+        return QString();
+    }
+
+    // always error out instead of returning a truncated string when the
+    // buffer is too small - avoids race condition when the user changes their
+    // default browser between calls to AssocQueryString
+    auto flags = ASSOCF_VERIFY | ASSOCF_NOTRUNCATE;
+
+    if (queryType == AQT_PROTOCOL)
+    {
+        // ASSOCF_IS_PROTOCOL was introduced in Windows 8
+        if (IsWindows8OrGreater())
+        {
+            flags |= ASSOCF_IS_PROTOCOL;
+        }
+        else
+        {
+            return QString();
+        }
```

Comment:
The documentation doesn't specify, it just says this flag was introduced in Windows 8. I assumed it was possible that it would be ignored, causing the protocol to be looked up as a file extension.

Theoretically the lookup would always fail even in that case since it doesn't have a period in front of it, but I figured it would be more correct and efficient to just error out early. COM calls aren't especially cheap.

Reply:
Alright, that's ok by me!